docs/early-access/devcontainers.md

```diff
@@ -0,0 +1,292 @@
+  agent_id         = coder_agent.dev.id
+  workspace_folder = "/home/coder/my-repository"
+}
+```
+
+> [!NOTE]
+>
+> The `workspace_folder` attribute must specify the location of the dev
+> container's workspace and should point to a valid project folder containing a
+> `devcontainer.json` file.
+
+> [!TIP]
+>
+> Consider using the [`git-clone`](https://registry.coder.com/modules/git-clone)
+> module to ensure your repository is cloned into the workspace folder and ready
+> for automatic startup.
+
+### Complete Template Example
+
+Here's a simplified template example that enables the dev containers
+integration:
+
+```terraform
+terraform {
+  required_providers {
+    coder  = { source = "coder/coder" }
+    docker = { source = "kreuzwerker/docker" }
+  }
+}
+
+provider "coder" {}
+data "coder_workspace" "me" {}
+data "coder_workspace_owner" "me" {}
+
+resource "coder_agent" "dev" {
+  arch                    = "amd64"
+  os                      = "linux"
+  startup_script_behavior = "blocking"
+  startup_script          = "sudo service docker start"
+  shutdown_script         = "sudo service docker stop"
+  # ...
+}
+
+module "devcontainers-cli" {
+  count    = data.coder_workspace.me.start_count
+  source   = "dev.registry.coder.com/modules/devcontainers-cli/coder"
+  agent_id = coder_agent.dev.id
+}
+
+resource "coder_devcontainer" "my-repository" {
+  count            = data.coder_workspace.me.start_count
+  agent_id         = coder_agent.dev.id
+  workspace_folder = "/home/coder/my-repository"
+}
+
+resource "docker_container" "workspace" {
+  count = data.coder_workspace.me.start_count
+  image = "codercom/oss-dogfood:latest"
+  env = [
+    "CODER_AGENT_DEVCONTAINERS_ENABLE=true",
+    # ...
+  ]
+  # ...
+}
+```
+
+## Working with Dev Containers
+
+### SSH Access
+
+You can SSH into your dev container directly using the Coder CLI:
+
+```console
+coder ssh --container keen_dijkstra my-workspace
+```
+
+> [!NOTE]
+>
+> SSH access is not yet compatible with the `coder config-ssh` command for use
+> with OpenSSH. You would need to manually modify your SSH config to include the
+> `--container` flag in the `ProxyCommand`.
+
+### Web Terminal Access
+
+Once your workspace and dev container are running, you can use the web terminal
+in the Coder interface to execute commands directly inside the dev container.
+
+### IDE Integration (VS Code)
+
+You can open your dev container directly in VS Code by:
+
+1. Selecting "Open in VS Code Desktop" from the Coder web interface
+2. Using the Coder CLI with the container flag:
+
+```console
+coder open vscode --container keen_dijkstra my-workspace
+```
+
+While optimized for VS Code, other IDEs with dev containers support may also
+work.
+
+### Port Forwarding
+
+During the early access phase, port forwarding is limited to ports defined via
+[`appPort`](https://containers.dev/implementors/json_reference/#image-specific)
+in your `devcontainer.json` file.
```

Comment:
Suggest specifically calling out plans for `forwardPorts` at a later date, and potentially linking to a relevant GH issue.